docs/tutorials/grid5000/virt/tuto_grid5000_virt.py

```diff
@@ -19,39 +22,31 @@ def range_mac(mac_start, mac_end, step=1):
         ip = ['10'] + [str(int(i, 2)) for i in mac.bits().split('-')[-3:]]
         yield str(mac).replace('-', ':'), '.'.join(ip)
 
-provider_conf = {
-    "job_type": "allow_classic_ssh",
-    "job_name": "test-non-deploy",
-    "walltime": "1:00:00",
-    "resources": {
-        "machines": [{
-            "role": "compute",
-            "cluster": "parasilo",
-            "nodes": PMS,
-            "primary_network": "n1",
-            "secondary_networks": []
-        }],
-        "networks": [{
-            "id": "n1",
-            "type": "prod",
-            "role": "my_network",
-            "site": "rennes",
-         }, {
-            "id": "not_linked_to_any_machine",
-            "type": "slash_22",
-            "role": "my_subnet",
-            "site": "rennes",
-         }]
-    }
-}
+# claim the resources
+prod = NetworkConfiguration(id="n1",
+                            type="prod",
```

Comment:
:+1:

Reply:
Actually ...
this field will be also validated: 
- automatically by `from_dictionnary` method
- or manually calling `.finalize()` after crafting the Configuration object.

So I'm for relying on this validation as much as possible. In the above case `type` is validated as an enum (see schema.py). An error (explicit enough IMO) will be thrown to the client in case of faulty value.